packages/gatsby/cache-dir/app.js

```diff
@@ -34,23 +34,38 @@ apiRunnerAsync(`onClientEntry`).then(() => {
     .then(res => res.json())
     .then(services => {
       if (services.developstatusserver) {
+        let isRestarting = false
         const parentSocket = io(
           `http://${window.location.hostname}:${services.developstatusserver.port}`
         )
 
-        parentSocket.on(`develop:needs-restart`, msg => {
+        parentSocket.on(`structured-log`, msg => {
           if (
-            window.confirm(
-              `The develop process needs to be restarted for the changes to ${msg.dirtyFile} to be applied.\nDo you want to restart the develop process now?`
-            )
+            !isRestarting &&
+            msg.type === `LOG_ACTION` &&
+            msg.action.type === `DEVELOP` &&
+            msg.action.payload === `RESTART_REQUIRED`
           ) {
-            parentSocket.once(`develop:is-starting`, msg => {
-              window.location.reload()
-            })
-            parentSocket.once(`develop:started`, msg => {
-              window.location.reload()
-            })
-            parentSocket.emit(`develop:restart`)
+            if (
```

Comment:
Nit: `window.confirm` is blocking. You could put this in the same conditional block and drop the extra indentation block... :shrug: